src/server/src/main/java/io/cassandrareaper/jmx/JmxProxyImpl.java

```diff
@@ -945,61 +943,20 @@ public void close() {
-   *
-   * <p>
-   * Compares two version strings.
-   *
-   * <p>
-   * Use this instead of String.compareTo() for a non-lexicographical comparison that works for version strings. e.g.
-   * "1.10".compareTo("1.6").
+   * Compares two Cassandra versions using classes provided by the Datastax Java Driver.
    *
    * @param str1 a string of ordinal numbers separated by decimal points.
    * @param str2 a string of ordinal numbers separated by decimal points.
-   * @return The result is a negative integer if str1 is _numerically_ less than str2. The result is a positive integer
-   *      if str1 is _numerically_ greater than str2. The result is zero if the strings are _numerically_ equal. It does
-   *      not work if "1.10" is supposed to be equal to "1.10.0".
+   * @return The result is a negative integer if str1 is _numerically_ less than str2. The result is
+   *     a positive integer if str1 is _numerically_ greater than str2. The result is zero if the
+   *     strings are _numerically_ equal. It does not work if "1.10" is supposed to be equal to
+   *     "1.10.0".
    */
-  static Integer versionCompare(String str1, String str2) throws ReaperException {
-    try {
-      String cleanedUpStr1 = str1.split(" ")[0].replaceAll("[-_~]", ".");
-      String cleanedUpStr2 = str2.split(" ")[0].replaceAll("[-_~]", ".");
-      String[] parts1 = cleanedUpStr1.split("\\.");
-      String[] parts2 = cleanedUpStr2.split("\\.");
-      int idx = 0;
-      // set index to first non-equal ordinal or length of shortest version string
-      while (idx < parts1.length && idx < parts2.length) {
-        try {
-          Integer.parseInt(parts1[idx]);
-          Integer.parseInt(parts2[idx]);
-        } catch (NumberFormatException ex) {
-          if (idx == 0) {
-            throw ex; // just comparing two non-version strings should fail
-          }
-          // first non integer part, so let's just stop comparison here and ignore the res
-          idx--;
-          break;
-        }
-        if (parts1[idx].equals(parts2[idx])) {
-          idx++;
-          continue;
-        }
-        break;
-      }
-      // compare first non-equal ordinal number
-      if (idx < parts1.length && idx < parts2.length) {
-        int diff = Integer.valueOf(parts1[idx]).compareTo(Integer.valueOf(parts2[idx]));
-        return Integer.signum(diff);
-      } else {
-        // the strings are equal or one string is a substring of the other
-        // e.g. "1.2.3" = "1.2.3" or "1.2.3" < "1.2.3.4"
-        return Integer.signum(parts1.length - parts2.length);
-      }
-    } catch (RuntimeException ex) {
-      LOG.error("failed comparing strings for versions: '{}' '{}'", str1, str2);
-      throw new ReaperException(ex);
-    }
+  static Integer versionCompare(String str1, String str2) {
+    VersionNumber version1 = com.datastax.driver.core.VersionNumber.parse(str1);
+    VersionNumber version2 = com.datastax.driver.core.VersionNumber.parse(str2);
```

Comment:
trivial: not too sure why the fully qualified classes here when it's already imported.

Reply:
My bad, I'll fix this before merging.
And yes, feels great to remove the boilerplate :)